docs/DeveloperGuide.md

```diff
@@ -1,37 +1,1036 @@
+2. Clone your fork to your machine.
+3. Set up your local repo in your IDE.
+    - Ensure that the project in your IDE is configured to run on Java JDK version 11. 
+    - A guide on setting your project to use JDK 11 in your IntelliJ IDEA IDE can be found 
+[here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).
+4. Run `Main.java`. If you have set up your environment correctly, you should see the following
+output in your terminal: 
+
+    ```
+    ======================================================================
+     __        __        _    ___ _   _ 
+     \ \      / /__ _ __| | _|_ _| |_| |
+      \ \ /\ / / _ \ '__| |/ /| || __| |
+       \ V  V /  __/ |  |   < | || |_|_|
+        \_/\_/ \___|_|  |_|\_\___|\__(_)
+
+    Welcome to WerkIt!, your personal exercise planner.
+    ----------------------------------------------------------------------
+    Checking for required directory and files...
+    - The required data directory was not found. It will be created.
+    - The WerkIt! resource directory has been created in
+      your terminal's current working directory.
+
+    - The exercise file was not found. It will be created.
+    - The exercise file 'exercises.txt' has been created in
+      the WerkIt! resource directory.
+
+    - The workout file was not found. It will be created.
+    - The workout file 'workouts.txt' has been created in
+      the WerkIt! resource directory.
+
+    - The plan file was not found. It will be created.
+    - The plan file 'plans.txt' has been created in
+      the WerkIt! resource directory.
+
+    - The schedule file was not found. It will be created.
+    - The schedule file 'schedule.txt' has been
+      created in the WerkIt! resource directory.
+
+    Loading saved file data...
+    - Exercises file	OK!
+    ----------------------------------------------------------------------
+    Now then, what can I do for you today?
+    (Need help? Type 'help' for a guide!)
+    ----------------------------------------------------------------------
+    >
+    ```
+5. Type `exit` to exit the program.
+
+You are now ready to begin developing!
+
+## Design 
+### Overview
+### Architecture Diagram
+
+![Architecture-Diagram](high-level-diagram/architecture_diagram.png)
+
+The features of WerkIt! are split and grouped into 5 main features:
+1. [Exercise-related features](#exercise-related-features)
+2. [Workout-related features](#workout-related-features)
+3. [Plan-related features](#plan-related-features)
+4. [Schedule-related features](#schedule-related-features)
+5. [Search-related features](#search-related-features)
+
+### Exercise-related features
+_to be updated_
+
+---
+
+### Workout-related features
+_to be updated_
+
+---
+
+### Plan-related features
+_to be updated_
+
+---
+
+### Schedule-related features
+_to be updated_
+
+---
+
+### Search-related features
+
+Below is a class diagram of the search-related features:
+![SearchUML](uml/classDiagrams/images/SearchClass.png)
```

Comment:
Are there any navigability between classes i.e. unidirection from UI to WerkIt?
Could the classes be arranged in a better format?

![image](https://user-images.githubusercontent.com/69353464/160963116-857f394c-505b-4856-8d83-5bb28ab6ac73.png)

Reply:
I'd like to chime in on this too. Are there any special considerations in placing Command and SearchCommand beofre WerkIt? Would it be better to have the flow follow a sequential order from left to right?

Reply:
Would like to echo this comment: is it possible to perhaps put the classes at different levels as per `listWorkout.puml`, instead of all in one line?